source_gen/lib/src/builder.dart

```diff
@@ -334,6 +342,19 @@ Stream<GeneratedOutput> _generate(
   }
 }
 
+Future<bool> _hasAnyTopLevelAnnotations(
+    AssetId input, Resolver resolver) async {
+  final parsed = await resolver.compilationUnitFor(input);
+  final annotatableObjects = [
```

Comment:
[nit] I wouldn't collect these to a list before we iterate over them.

Either 2 for loops, or `parsed.directives.followedBy(parsed.declarations)` would avoid an extra list.

Reply:
Going with two loops, I can't use `followedBy` without casting the iterable first or creating an empty iterable of the right type followed by the other two both of which were pretty gross.